test/common/stats/thread_local_store_test.cc

```diff
@@ -1567,6 +1567,122 @@ TEST_F(HistogramTest, ParentHistogramBucketSummary) {
             parent_histogram->bucketSummary());
 }
 
+class SinkPredicatesTest : public SinkPredicates {
+public:
+  ~SinkPredicatesTest() override = default;
+  absl::flat_hash_set<std::string>& sinkedStatNames() { return sinked_stat_names_; }
+  bool includeCounter(const Counter&) override { return false; }
+  bool includeGauge(const Gauge&) override { return false; }
+  bool includeTextReadout(const TextReadout&) override { return false; }
+  bool includeHistogram(const Histogram& histogram) override {
+    return sinked_stat_names_.find(histogram.tagExtractedName()) != sinked_stat_names_.end();
+  }
+
+private:
+  absl::flat_hash_set<std::string> sinked_stat_names_;
+};
+
+TEST_F(HistogramTest, ForEachHistogram) {
+  std::vector<std::reference_wrapper<Stats::Histogram>> histograms;
+
+  const size_t num_stats = 11;
+  for (size_t idx = 0; idx < num_stats; ++idx) {
+    auto stat_name = absl::StrCat("histogram.", idx);
+    histograms.emplace_back(
+        store_->histogramFromString(stat_name, Stats::Histogram::Unit::Unspecified));
+  }
+  EXPECT_EQ(histograms.size(), 11);
+
+  size_t num_histograms = 0;
+  size_t num_iterations = 0;
+  store_->forEachHistogram([&num_histograms](std::size_t size) { num_histograms = size; },
+                           [&num_iterations](Stats::ParentHistogram&) { ++num_iterations; });
+  EXPECT_EQ(num_histograms, 11);
+  EXPECT_EQ(num_iterations, 11);
+
+  Histogram& deleted_histogram = histograms[4];
+
+  // Verify that rejecting histograms removes them from the iteration set.
+  envoy::config::metrics::v3::StatsConfig stats_config_;
+  stats_config_.mutable_stats_matcher()->set_reject_all(true);
+  store_->setStatsMatcher(std::make_unique<StatsMatcherImpl>(stats_config_, symbol_table_));
+  num_histograms = 0;
+  num_iterations = 0;
+  store_->forEachHistogram([&num_histograms](std::size_t size) { num_histograms = size; },
+                           [&num_iterations](Stats::ParentHistogram&) { ++num_iterations; });
+  EXPECT_EQ(num_histograms, 0);
+  EXPECT_EQ(num_iterations, 0);
+
+  // Verify that we can access the local reference without a crash.
+  (void)deleted_histogram.unit();
```

Comment:
why not just EXPECT_EQ on this? We typically don't have this `(void)` cast in Envoy code for whatever reason.

Reply:
Done